controllers/resources/manager.go

```diff
@@ -123,8 +127,14 @@ func (m *manager) generateRemediationCR(name string, healthCheckOwnerRef *metav1
 	// can't go wrong, we already checked for correct spec
 	templateSpec, _, _ := unstructured.NestedMap(template.Object, "spec", "template", "spec")
 	unstructured.SetNestedField(remediationCR.Object, templateSpec, "spec")
+	crName := name
+	if annotationutils.HasMultipleTemplatesAnnotation(template) {
+		//first part would be the node name and second part is a unique suffix seperated by '_'
+		crName = generateUniqueRemediationName(name)
```

Comment:
also, review the comment above. With `generateUniqueRemediationName` the suffix is separated by a dash, not underscore.

Reply:
>no need to re-invent the wheel, please use remediationCR.SetGenerateName(name)

Maybe I'm misunderstanding how `GenerateName` works, IIUC it's  is just another field in the metadata.
If that is the case:
- Setting to remediations of same kind (same node) of different templates does not make any sense because `GenerateName` will be identical in both (i.e node name)
- Even if we'll use different value for  `GenerateName` we still won't be able to create 2 remediation because their unique identifier will be the same (same kind, same ns and same name).

Reply:
https://kubernetes.io/docs/reference/using-api/api-concepts/#generated-values